zipline/finance/performance/tracker.py

```diff
@@ -388,6 +400,20 @@ def handle_market_close(self, dt, data_portal):
                 benchmark_value,
                 account.leverage)
 
+        if calculate_position_weights:
+            portfolio = self.get_portfolio(performance_needs_update=False)
+            position_weights = portfolio.current_portfolio_weights()
+
+            # For each day of the simulation, check to see if any futures were
+            # held. If so, convert them to continuous futures.
+            convert_futures = partial(
+                portfolio.asset_for_history_call, date=portfolio.current_date,
+            )
+            position_weights.index = list(
+                map(convert_futures, position_weights.index),
+            )
+            self.position_weights.append(dict(position_weights))
```

Comment:
Had you investigated other structures for storing the position weights?
e.g. a dict of numpy arrays or a sparse `DataFrame`, which would add a new column for each asset as a position is held?

We convert this data to a `DataFrame` anyway in `_calculate_rolling_expected_shortfall`.

I haven't sketched it out, but I think the overhead of creating and storing a dict with the keys for each position for every backtest day may be more costly than storing the data in a sparse `DataFrame`.

The caveat with storing as a `DataFrame` is that as each position is held a new column will be added throughout the lifetime of the backtest, with many possible rows where there is no value; however the sparse representation should mitigate that growth.

Reply:
> Had you investigated other structures for storing the position weights?

No I hadn't really considered other structures, mainly due to your point about a DataFrame being mostly empty for a lot of values. But I had not thought of using a sparse `DataFrame`, so I can investigate to see if that would help.

Reply:
@ehebert Sketched out a possibility here: https://github.com/quantopian/zipline/pull/1952

Reply:
Addressed this in #1952, but TLDR is I think what we have is a better option overall.